pkg/controllers/vdb/stopsc_reconciler.go

```diff
@@ -0,0 +1,98 @@
+/*
+ (c) Copyright [2021-2024] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package vdb
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	vapi "github.com/vertica/vertica-kubernetes/api/v1"
+	"github.com/vertica/vertica-kubernetes/pkg/cmds"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	"github.com/vertica/vertica-kubernetes/pkg/events"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/stopsc"
+	config "github.com/vertica/vertica-kubernetes/pkg/vdbconfig"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+type StopSubclusterReconciler struct {
+	Rec            config.ReconcilerInterface
+	Vdb            *vapi.VerticaDB // Vdb is the CRD we are acting on.
+	PRunner        cmds.PodRunner
+	Log            logr.Logger
+	PFacts         *PodFacts
+	SCName         string
+	InitiatorPodIP string
+	Dispatcher     vadmin.Dispatcher
+}
+
+func MakeStopSubclusterReconciler(rec config.ReconcilerInterface, log logr.Logger,
+	vdb *vapi.VerticaDB, scName string, pfacts *PodFacts, dispatcher vadmin.Dispatcher) controllers.ReconcileActor {
```

Comment:
I thought that will be the caller's responsibility. Also why do we need to stop subcluster before sandboxing? Based on the doc(https://docs.vertica.com/24.1.x/en/eon/subcluster-sandboxing/creating-sandboxes/) all subcluster's node must be UP in order to be sandboxed.

Reply:
We are relaxing that condition. I think vclusterops already can handle the case where we sandbox and the nodes are down. We want to control it because we want to give the ability to the user to wait for connections to go away before we sandbox. This is same reason why we drain for online upgrade.

Reply:
Also, we want this reconciler to be independent from the sandbox reconciler that Cai is working on.

Reply:
I see. So we stop subcluster in 2 places: before sandboxing in vdb controller and before unsandboxing in sandbox controller. Is that correct?

Reply:
Yes, that is correct.